### DISCUSSION TITLE
Add Bcrypt support.

### DISCUSSION THREAD

**Discussion by baek-sang** [samuel-lucas6/Geralt#1] [Ideas]
Any plan?

**Comment by samuel-lucas6** [comment]
Afraid not. bcrypt isn't in libsodium, only Argon2id, Argon2i, and scrypt are. Furthermore, bcrypt is weaker than Argon2 with decent parameters, and I intend to stick with one algorithm per purpose if possible.

Whilst bcrypt is stronger than Argon2 with lower parameters, it can't be used as a KDF, and there are weird restrictions on the password, so it's not very versatile.

**Comment by samuel-lucas6** [comment]
As an update, if libsodium [implements bscrypt](https://github.com/jedisct1/libsodium-doc/commit/08a9e0355ed57fefaf28af7c49ab6ad1c111b4ea), that will probably be added and encouraged for password hashing instead of Argon2.
